src/debugging/Inspector/scripts/generate_typescript_interfaces.py

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env python
+
+import sys
+import getopt
+
+def main(argv):
+    usage = 'Usage: generate_typescript_interfaces.py --inspector-scripts-path <path to JavaScriptCore/inspector/scripts> --combined-domains-path <path to CombinedDomains.json> --output-dir <result directory path>'
```

Comment:
1. --combined-domains-path - CombinedDomains.json is generated on JavaScriptCore build by this python script [generate-combined-inspector-json.py](https://github.com/NativeScript/webkit/blob/master/Source/JavaScriptCore/inspector/scripts/generate-combined-inspector-json.py)
It basiclly combines all the jsons that describe the different inspector domains. The location of CombinedDomains.json is under Derived Resources/JavaScriptCore. On my machine the location is /Users/koeva/Library/Developer/Xcode/DerivedData/WebKit-fxkfueihzasqpgepvxujfkvdvzlz/Build/Products/Debug/DerivedSources/JavaScriptCore/CombinedDomains.js
2. --inspector-scripts-path - The CombinedDomains.json is parsed into modules that describes every domain, its type defs, commands, events etc. The parsing logic is located under ios-runtime/src/webkit/Source/JavaScriptCore/inspector/scripts. In order to reuse that parsing logic (and not write our own) we import those scripts into our python generator 
3. --output-dir is the directory where the generated typescript file should be positioned

We've decided that this python script should not be run as part of our build, but on demand. The general usage (again on my machine) is 
python generate_typescript_interfaces.py --inspector-scripts-path "/Users/koeva/work/tns/ios-runtime/src/webkit/Source/JavaScriptCore/inspector/scripts" --combined-domains-path "/Users/koeva/Library/Developer/Xcode/DerivedData/WebKit-fxkfueihzasqpgepvxujfkvdvzlz/Build/Products/Debug/DerivedSources/JavaScriptCore/CombinedDomains.js" --output-dir "/Users/koeva/work/temp" 

I am not sure that this is the best approach so if you have some other idea about running this script or the parameters it accepts, please share it :)